webapp/src/components/issue_attribute_selector.tsx

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect} from 'react';
+import ReactSelect, {OnChangeValue} from 'react-select';
+import {Theme} from 'mattermost-redux/types/preferences';
+
+import {getStyleForReactSelect} from 'src/utils/styles';
+import {usePrevious} from 'src/hooks/use_previous';
+
+import Setting from './setting';
+
+type PropTypes = {
+    isMulti: boolean;
+    projectName: string;
+    theme: Theme;
+    label: string;
+    onChange: (value: OnChangeType) => void;
+    loadOptions: () => Promise<Array<SelectionType>>,
+    selection: OnChangeType;
+};
+
+const IssueAttributeSelector = ({isMulti, projectName, theme, label, onChange, loadOptions, selection}: PropTypes) => {
+    const [options, setOptions] = useState<SelectionType[]>([]);
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [error, setError] = useState<string>('');
+
+    useEffect(() => {
+        if (projectName) {
+            loadSelectOptions();
+        }
+    }, []);
```

Comment:
Should this instead take `projectName` as dep?

```suggestion
    useEffect(() => {
        if (projectName) {
            loadSelectOptions();
        }
    }, [projectName]);
```